pymine_net/types/vector.py

```diff
@@ -1,30 +1,27 @@
-from typing import TypeVar
+from dataclasses import dataclass
+from typing import Generic, TypeVar
 
-T = TypeVar("T")
+T = TypeVar("T", bound=float)
 
 
-class Vector3:
+@dataclass(slots=True)
+class Vector3(Generic[T]):
     """
     Stores three numeric values: x, y, z.
     - Used for position and movement data in the Player class.
     """
 
-    __slots__ = ("x", "y", "z")
+    x: T
+    y: T
+    z: T
 
-    def __init__(self, x: T, y: T, z: T):
-        self.x = x
-        self.y = y
-        self.z = z
 
-
-class Rotation:
+@dataclass(slots=True)
```

Comment:
The slots kwarg is is a 3.10+ feature iirc, please revert back to using a regular class.